ott/geometry/grid.py

```diff
@@ -354,3 +364,54 @@ def tree_unflatten(cls, aux_data, children):
     return cls(
         x=children[0], cost_fns=children[1], epsilon=children[2], **aux_data
     )
+
+  def to_LRCGeometry(
+      self,
+      scale: float = 1.0,
+      **kwargs: Any,
+  ) -> low_rank.LRCGeometry:
+    """Converts grid to low-rank geometry.
+
+    Conversion is carried out by taking advantage of the fact that the true cost
+    matrix of a grid geometry is a sum of kronecker products of local cost
+    matrices (for each dimension) with matrice of 1's (both on left and right
+    sides) of varying dimension. Each of the matrices in that sum can be
+    factorized if each of these cost matrices can be factorized, which we do
+    by forcing a conversion to a low rank geometry object.
+
+    Args:
+      scale: Value used to rescale the factors of the low-rank geometry.
+        Useful when this geometry is used in the linear term of fused GW.
+
+    Returns:
+      Low-rank geometry.
+    """
+    cost_1 = []
+    cost_2 = []
+    for dimension, geom in enumerate(self.geometries):
+      geom = geom.to_LRCGeometry(rank=0, scale=scale, **kwargs)
```

Comment:
I'd add a comment why `rank=0` is being passed.

Reply:
good idea.